### PR TITLE
profilerecorder: gracefully skip containers that did not record anything

### PR DESCRIPTION
**Pull request by jhrozek**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
This is mostly concerning SELinux because even containers that don't
seemingly do anything at least record some syscalls, e.g. exec.

But with SELinux, if a container only performs actions that are allowed
by the container policy in the first place, no AVCs would be generated.
In that case, the call to get AVCs would return with a rather generic
error, which would then be retried by the profilerecording reconciler.
As a consequence, if a pod had multiple containers and the one with no
AVCs was collected before others, the profilerecorder would never reach
the other containers.

Instead, let's return an error matching the situation and handle the
absence of AVCs gracefully, by emptying whatever there might be in the
grpc cache and moving on to the next container.

#### Which issue(s) this PR fixes:
None

#### Does this PR have test?
Yes

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by jhrozek** [comment]
Looks like the OLM tests are broken right now:
```
+ operator-sdk olm install --version v0.18.2 --timeout 6m
time="2022-12-13T14:07:45Z" level=info msg="Fetching CRDs for version \"v0.18.2\""
time="2022-12-13T14:07:45Z" level=info msg="Fetching resources for resolved version \"v0.18.2\""
2022/12/13 14:07:45 protocol error: received *http.http2GoAwayFrame before a SETTINGS frame
time="2022-12-13T14:07:45Z" level=fatal msg="Failed to install OLM version \"v0.18.2\": failed to get resources: failed to fetch CRDs: request failed: failed GET 'https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.18.2/crds.yaml': Get \"[https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.18.2/crds.yaml\](https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.18.2/crds.yaml/)": connection error: PROTOCOL_ERROR"
```

**Comment by jhrozek** [comment]
olm tests and the regular e2e tests now passed, but the fedora e2e failed twice with the same symptom. Going to look into that.
I could swear that the tests passed for me locally earlier in vagrant..

**Review by ccojocar — APPROVED** [review]

**Comment by jhrozek** [comment]
Rebased atop current main

**Review by ccojocar — APPROVED** [review]

**Comment by jhrozek** [comment]
Rebased atop the flatcar test fixes again

**Comment by jhrozek** [comment]
hmm please hold off with a follow up review before I triage the test failures..

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go
Comment (jhrozek) [inline internal/pkg/daemon/bpfrecorder/bpfrecorder.go:726]: FWIW, this was the isse, we generated longer statuses and iterated over them, but inside the for loop we would have still assigned `containerStatus := pod.Status.ContainerStatuses[c]`.

Comment (jhrozek) [reply internal/pkg/daemon/bpfrecorder/bpfrecorder.go:726]: The upcoming patch version changes to `containerStatus := statuses[c]`

**Comment by jhrozek** [comment]
@ccojocar Hi, I'm sorry for the super long delay in getting back to this PR. I got pulled into other directions at work. Could I please get another `lgtm`? I added the reason the patch was initally failing into the PR as a comment, because of course you can't remember the PR anymore :-)
There are no other changes.

**Review by ccojocar — APPROVED** [review]

**Pull request merged**
